substrate/pkg/controller/substrate/cluster/instanceprofile.go

```diff
@@ -186,6 +185,91 @@ func desiredRolesFor(substrate *v1alpha1.Substrate) []role {
+						"ec2:DescribeInternetGateways",
+						"ec2:DescribeVpcs",
+						"ec2:DescribeNetworkInterfaces",
+						"ec2:DescribeTags",
+						"ec2:GetCoipPoolUsage",
+						"ec2:DescribeCoipPools",
+						"ec2:CreateLaunchTemplateVersion",
+						"ec2:DeleteLaunchTemplate",
+						"ec2:DescribeLaunchTemplateVersions",
+						"ec2:AuthorizeSecurityGroupIngress",
+						"ec2:RevokeSecurityGroupIngress",
+						"ec2:CreateSnapshot",
+						"ec2:AttachVolume",
+						"ec2:DetachVolume",
+						"ec2:ModifyVolume",
+						"ec2:DescribeVolumes",
+						"ec2:DescribeVolumesModifications",
+						"ec2:DeleteTags",
+						"ec2:CreateVolume",
+						"ec2:DeleteVolume",
+						"ec2:DeleteSnapshot",
+						"elasticloadbalancing:DescribeLoadBalancers",
+						"elasticloadbalancing:DescribeLoadBalancerAttributes",
+						"elasticloadbalancing:DescribeListeners",
+						"elasticloadbalancing:DescribeListenerCertificates",
+						"elasticloadbalancing:DescribeSSLPolicies",
+						"elasticloadbalancing:DescribeRules",
+						"elasticloadbalancing:DescribeTargetGroups",
+						"elasticloadbalancing:DescribeTargetGroupAttributes",
+						"elasticloadbalancing:DescribeTargetHealth",
+						"elasticloadbalancing:DescribeTags",
+						"elasticloadbalancing:CreateLoadBalancer",
+						"elasticloadbalancing:CreateTargetGroup",
+						"elasticloadbalancing:CreateListener",
+						"elasticloadbalancing:DeleteListener",
+						"elasticloadbalancing:CreateRule",
+						"elasticloadbalancing:DeleteRule",
+						"elasticloadbalancing:AddTags",
+						"elasticloadbalancing:RemoveTags",
+						"elasticloadbalancing:ModifyLoadBalancerAttributes",
+						"elasticloadbalancing:SetIpAddressType",
+						"elasticloadbalancing:SetSecurityGroups",
+						"elasticloadbalancing:SetSubnets",
+						"elasticloadbalancing:DeleteLoadBalancer",
+						"elasticloadbalancing:ModifyTargetGroup",
+						"elasticloadbalancing:ModifyTargetGroupAttributes",
+						"elasticloadbalancing:DeleteTargetGroup",
+						"elasticloadbalancing:RegisterTargets",
+						"elasticloadbalancing:DeregisterTargets",
+						"elasticloadbalancing:SetWebAcl",
+						"elasticloadbalancing:ModifyListener",
+						"elasticloadbalancing:AddListenerCertificates",
+						"elasticloadbalancing:RemoveListenerCertificates",
+						"elasticloadbalancing:ModifyRule",
+						"iam:CreateRole",
+						"iam:PassRole",
+						"iam:AddRoleToInstanceProfile",
+						"iam:CreateInstanceProfile",
+						"iam:AttachRolePolicy",
+						"iam:RemoveRoleFromInstanceProfile",
+						"iam:DeleteInstanceProfile",
+						"iam:DetachRolePolicy",
+						"iam:DeleteRole",
+						"iam:TagRole",
+						"iam:GetRole",
+						"iam:GetInstanceProfile",
+						"iam:CreateServiceLinkedRole",
+						"iam:ListServerCertificates",
+						"iam:GetServerCertificate",
+						"shield:GetSubscriptionState",
+						"shield:DescribeProtection",
+						"shield:CreateProtection",
+						"shield:DeleteProtection",
+						"waf-regional:GetWebACL",
```

Comment:
Where did all these come from?

Reply:
These are the permissions required by ELB controller and CSI driver, I got these from their policy files.